src/context.rs

```diff
@@ -2197,6 +2203,87 @@ impl<'a> ContextWriter<'a> {
     );
   }
 
+  // Based on https://aomediacodec.github.io/av1-spec/#cdf-selection-process
+  // Used to decide the cdf (context) for txfm_split
+  fn get_above_tx_width(
+    &self, bo: TileBlockOffset, _bsize: BlockSize, _tx_size: TxSize,
+  ) -> usize {
+    let has_above = bo.0.y > 0;
+    // TODO: if ( tx block is topmost within a partition )
+    if !has_above {
+      return 64;
+    }
+    let above_blk = self.bc.blocks.above_of(bo);
+    if above_blk.skip && above_blk.is_inter() {
+      return above_blk.bsize.width();
+    };
+
+    self.bc.above_tx_context[bo.0.x] as usize
+  }
+
+  fn get_left_tx_height(
+    &self, bo: TileBlockOffset, _bsize: BlockSize, _tx_size: TxSize,
+  ) -> usize {
+    let has_left = bo.0.x > 0;
+    // TODO: if ( tx block is leftmost within a partition )
+    if !has_left {
+      return 64;
+    }
+    let left_blk = self.bc.blocks.left_of(bo);
+    if left_blk.skip && left_blk.is_inter() {
+      return left_blk.bsize.height();
+    };
+
+    self.bc.left_tx_context[bo.y_in_sb()] as usize
+  }
+
+  fn txfm_partition_context(
+    &self, bo: TileBlockOffset, bsize: BlockSize, tx_size: TxSize,
+  ) -> usize {
+    // TODO: from 2nd level partition, must know whether the tx block is the topmost(or leftmost) within a partition
+    let above =
+      (self.get_above_tx_width(bo, bsize, tx_size) < tx_size.width()) as usize;
+    let left = (self.get_left_tx_height(bo, bsize, tx_size) < tx_size.height())
+      as usize;
+
+    let mut category = TXFM_PARTITION_CONTEXTS;
+    let max_tx_size: TxSize = bsize.tx_size().sqr_up();
+
+    if max_tx_size >= TxSize::TX_8X8 {
+      category = (tx_size.sqr_up() != max_tx_size
+        && max_tx_size > TxSize::TX_8X8) as usize
+        + (TxSize::TX_SIZES as usize - 1 - max_tx_size as usize) * 2;
+    }
+    debug_assert!(category != TXFM_PARTITION_CONTEXTS);
+
+    category * 3 + above + left
+  }
+
+  pub fn write_tx_size_inter(
+    &mut self, w: &mut dyn Writer, bo: TileBlockOffset, bsize: BlockSize,
+    tx_size: TxSize, txfm_split: bool,
+  ) {
+    debug_assert!(self.bc.blocks[bo].is_inter());
+    debug_assert!(bsize > BlockSize::BLOCK_4X4);
+    debug_assert!(!tx_size.is_rect() || bsize.is_rect_tx_allowed());
+
+    let ctx = self.txfm_partition_context(bo, bsize, tx_size);
+
+    symbol_with_update!(
+      self,
+      w,
+      txfm_split as u32,
+      &mut self.fc.txfm_partition_cdf[ctx]
+    );
+
+    if !txfm_split {
+      self.bc.update_tx_size_context(bo, bsize, tx_size, false);
+    }
+
+    // TODO: if txfm_split == true, call write_tx_size_inter() for each of tx block in a partition
+    // if txfm_split {}
+  }
```

Comment:
maybe make this todo an assert (e.g. by else {unimplemented!()})

Reply:
Good idea!

Reply:
Done.